cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataSubscriberService.java

```diff
@@ -112,7 +123,7 @@
   MetadataSubscriberService(CConfiguration cConf, MessagingService messagingService,
                             DatasetFramework datasetFramework, TransactionSystemClient txClient,
                             MetricsCollectionService metricsCollectionService,
-                            MetadataStore metadataStore,
+                            AuditMetadataStorage metadataStorage,
```

Comment:
Rather than letting services individually decide whether they want audit support or not, I think it would be cleaner if  `AuditMetadataStorage` is returned by `MetadataStorageProvider`. This way if we want to turn off auditing, then we can do it in one place.

Reply:
good point